### PR TITLE
Add BIO memory buffer utility to memory_util

### LINKED ISSUE #92: Add BIO memory buffer utility to memory_util

### PR DESCRIPTION
**Pull request by ZacharyEspiritu**
Resolves #92. 

Used to parse RSA and EC PEM-encoded keys.

### CODE REVIEW

File: src/bridge/memory_util/openssl_bio.cc [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+#include "src/backing/status/status.h"
+#include "src/backing/status/status_or.h"
+#include "src/bridge/memory_util/openssl_structs.h"
+
+namespace kmsengine {
+namespace bridge {
+
+StatusOr<OpenSslBio> MakeOpenSslMemoryBufferBio(const void *data, int length) {
+  BIO *pem_stream = BIO_new_mem_buf(data, length);
+  if (pem_stream == nullptr) {
```
Comment (ajfeldma-google) [inline src/bridge/memory_util/openssl_bio.cc]: There isn't anything PEM-specific about BIOs, right? Maybe make the variable names generic?

File: src/bridge/memory_util/openssl_bio_test.cc [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  "ozo2t60EG8L0561g13R29LvMR5hyvGZlGJpmn65+A4xHXInJYiPuKzrKUnApeLZ+\n"
+  "vw1HocOAZtWK0z3r26uA8kQYOKX9Qt/DbCdvsF9wF8gRK0ptx9M6R13NvBxvVQAp\n"
+  "fc9jB9nTzphOgM4JiEYvlV8FLhg9yZovMYd6Wwf3aoXK891VQxTr/kQYoq1Yp+68\n"
+  "i6T4nNq7NWC+UNVjQHxNQMQMzU6lWCX8zyg3yH88OAQkUXIXKfQ+NkvYQ1cxaMoV\n"
+  "PpY72+eVthKzpMeyHkBn7ciumk5qgLTEJAfWZpe4f4eFZj/Rc8Y8Jj2IS5kVPjUy\n"
+  "wQIDAQAB\n"
+  "-----END PUBLIC KEY-----\n";
+
+TEST(OpenSslBioTest, MakeOpenSslMemoryBufferBioContainsExpectedBytes) {
+  auto public_key_bio_or = MakeOpenSslMemoryBufferBio(kRsaPublicKey,
```
Comment (ajfeldma-google) [inline src/bridge/memory_util/openssl_bio_test.cc]: Write out the variable types

File: src/bridge/memory_util/openssl_bio_test.cc [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+#include "src/testing_util/openssl_assertions.h"
+
+namespace kmsengine {
+namespace bridge {
+namespace {
+
+using ::testing::StrEq;
+using ::kmsengine::testing_util::IsOk;
+
+constexpr char kRsaPublicKey[] = "-----BEGIN PUBLIC KEY-----\n"
```
Comment (ajfeldma-google) [inline src/bridge/memory_util/openssl_bio_test.cc]: There isn't anything PEM-specific about BIOs, right? Maybe make the buffer contents generic?

**Review by ajfeldma-google — CHANGES_REQUESTED** [review]

File: src/bridge/memory_util/openssl_bio.cc (hunk 1 above)
Comment (ZacharyEspiritu) [reply src/bridge/memory_util/openssl_bio.cc]: Ah yes, this is an accidental holdover from the RSA code

**Review by ajfeldma-google — APPROVED** [review]

**Review by tonyliaoss — APPROVED** [review]

**Pull request merged**